python/cudf/cudf/core/mixins/binops.py

```diff
@@ -48,9 +48,22 @@
     },
 )
 
+# TODO: See if there is a better approach to these two issues: 1) The mixin
+# assumes a single standard parameter, whereas binops have two, and 2) we need
+# a way to determine reflected vs normal ops.
 
-def _is_reflected_op(op):
-    return op[2] == "r" and op != "__rshift__"
 
+def _binaryop(self, other, op: str):
+    "The core binary_operation function. Must be overridden by subclasses, "
+    "the default implementation raises a NotImplementedError."
+    raise NotImplementedError
 
-BinaryOperand._is_reflected_op = staticmethod(_is_reflected_op)
+
+def _check_reflected_op(op):
+    if (reflect := op[2] == "r" and op != "__rshift__") :
```

Comment:
Not sure if this could ever get called for `__round__`, `__reversed__`, or `__repr__`. Just a note of those other methods starting with `r`.

Alternatively we could use a lookup with a fixed mapping defined in the `utils.py` file like
```python
_reflected_ops = {
    "__radd__": "__add__",
    # ...
}
```
I lean a bit towards that because it requires no special cases and aligns well with the existing code paths for reversing comparisons (such as `__lt__` to `__gt__`) and mapping ufunc names.

Reply:
I am also not a huge fan of this method right now. I have done work to refactor this in other ways in downstream branches so I'm going to leave this as is in this PR, but it's definitely something that we need to do some work to improve. In general, reflected vs not-reflected ops are not the cleanest with this framework right now and I plan to try to improve that later.